File: Programmers_level1/13. 체육복.py
def solution(n, lost, reserve):
    answer = 0
    clothes = [1] * (n+1)
    
    for num in lost:
        clothes[num] -= 1
    for num in reserve:
        clothes[num] += 1
    
    for i in range(1, n+1):
        if clothes[i] == 2:
            if clothes[i-1] == 0:
                clothes[i] -= 1
                clothes[i-1] += 1
            elif i < n and clothes[i+1] == 0:
                clothes[i] -= 1
                clothes[i+1] += 1
    
    for i in range(1, n+1):
        if clothes[i] > 0:
            answer += 1
    
    # print(clothes)
    
    
    return answer

File: Programmers_level1/13. 체육복.swift
import Foundation

func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    var clothes = Array(repeating: 1, count: n+1)
    
    for l in lost {
        clothes[l] -= 1
    }
    
    for r in reserve {
        clothes[r] += 1
    }
    
    for i in 1..<clothes.count {
        if clothes[i] == 2 {
            if clothes[i-1] == 0 {
                clothes[i-1] = 1
                clothes[i] = 1
            } else if i < clothes.count - 1 && clothes[i+1] == 0 {
                clothes[i] = 1
                clothes[i+1] = 1
            }
        }
    }

    return clothes.filter {
        $0 > 0
    }.count - 1
}
